packages/ra-core/src/sideEffect/auth.ts

```diff
@@ -23,111 +23,115 @@ import {
-                    yield put(push(redirectTo || '/'));
-                } catch (e) {
-                    yield put({
-                        type: USER_LOGIN_FAILURE,
-                        error: e,
-                        meta: { auth: true },
-                    });
-                    const errorMessage = getErrorMessage(
-                        e,
-                        'ra.auth.sign_in_error'
-                    );
-                    yield put(showNotification(errorMessage, 'warning'));
-                }
-                break;
-            }
-            case USER_CHECK: {
-                try {
-                    yield call(authProvider, AUTH_CHECK, payload);
-                } catch (error) {
-                    yield call(authProvider, AUTH_LOGOUT);
-                    yield put(
-                        replace({
-                            pathname: (error && error.redirectTo) || '/login',
-                            state: { nextPathname: meta.pathName },
-                        })
-                    );
-                    const errorMessage = getErrorMessage(
-                        error,
-                        'ra.auth.auth_check_error'
-                    );
-                    yield put(showNotification(errorMessage, 'warning'));
-                }
-                break;
-            }
-            case USER_LOGOUT: {
-                yield put(
-                    push(
-                        (action.payload && action.payload.redirectTo) ||
-                            '/login'
-                    )
-                );
-                yield call(authProvider, AUTH_LOGOUT);
-                break;
-            }
-            case FETCH_ERROR:
-                try {
-                    yield call(authProvider, AUTH_ERROR, error);
-                } catch (e) {
-                    const nextPathname = yield select(currentPathnameSelector);
-                    yield call(authProvider, AUTH_LOGOUT);
-                    yield put(
-                        push({
-                            pathname: '/login',
-                            state: { nextPathname },
-                        })
-                    );
-                    yield put(hideNotification());
-                    yield put(
-                        showNotification(
-                            'ra.notification.logged_out',
-                            'warning'
-                        )
-                    );
-                }
-                break;
+export const handleLogin = (authProvider: AuthProvider) =>
+    function*(action) {
+        const { payload, meta } = action;
+        try {
+            yield put({ type: USER_LOGIN_LOADING });
+            const authPayload = yield call(authProvider, AUTH_LOGIN, payload);
+            yield put({
+                type: USER_LOGIN_SUCCESS,
+                payload: authPayload,
+            });
+            const redirectTo = yield meta.pathName ||
```

Comment:
Can we `yield` a string?
Would be more clear to me with `meta.pathName || yield select(...)`

Reply:
Wut why

Reply:
![react-admin-yield-ts](https://user-images.githubusercontent.com/1122076/61642447-2dd29b80-aca1-11e9-93d8-1e899f465a78.png)